src/hayhooks/server/routers/undeploy.py

```diff
@@ -1,22 +1,29 @@
-from typing import cast
-from fastapi import APIRouter, HTTPException
+from fastapi import APIRouter, HTTPException, Request
 from fastapi.routing import APIRoute
 from hayhooks.server.pipelines import registry
+from hayhooks.server.utils.deploy_utils import remove_pipeline_files
+from hayhooks.settings import settings
 
 router = APIRouter()
 
 
 @router.post("/undeploy/{pipeline_name}", tags=["config"])
-async def undeploy(pipeline_name: str):
+async def undeploy(pipeline_name: str, request: Request):
+    # Check if pipeline exists in registry
     if pipeline_name not in registry.get_names():
-        raise HTTPException(status_code=404)
+        raise HTTPException(status_code=404, detail=f"Pipeline '{pipeline_name}' not found")
 
-    new_routes = []
-    for route in router.routes:
-        route = cast(APIRoute, route)
-        if route.name != pipeline_name:
-            new_routes.append(route)
-
-    router.routes = new_routes
-    router.openapi_schema = None
+    # Remove pipeline from registry
     registry.remove(pipeline_name)
+
+    # Remove API routes for the pipeline
+    # YAML based pipelines have a run endpoint at /<pipeline_name>
+    # Wrapper based pipelines have a run endpoint at /<pipeline_name>/run
+    for route in request.app.routes:
+        if isinstance(route, APIRoute) and (route.path == f"/{pipeline_name}/run" or route.path == f"/{pipeline_name}"):
+            request.app.routes.remove(route)
+
+    # Remove pipeline files if they exist
+    remove_pipeline_files(pipeline_name, settings.pipelines_dir)
+
+    return {"success": True, "name": pipeline_name}
```

Comment:
If I understand correctly, this is the most significant change.

Could you please explain a bit more what the problem was and what we are doing now to fix it?

Reply:
You're right sorry ;)

So, `hayooks pipeline undeploy` was breaking [here](https://github.com/deepset-ai/hayhooks/blob/c039df3f6f9b142fe1e5865b62686edca2be4e6c/src/hayhooks/cli/pipeline.py#L115) due to the fact that the `/undeploy` endpoint was not returning a `name` key in JSON response.

After investigation, I found that it was also using a wrong way to remove a dynamic route in a FastAPI application and also wasn't properly supporting files-based pipelines (was not removing pipeline files, only removing pipeline from in-memory registry). 

So I've fixed those things and added tests for every aspect.